### PR TITLE
fix(router): set tabindex only if link is not null

### LINKED ISSUE #17616: routerLink adds tabindex even if the route is undefined
## I'm submitting a ...

<pre><code>
[ ] Regression (behavior that used to work and stopped working in a new release)
[X ] Bug report 
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
<div [routerLink]="undefined">has tabIndex</div>

## Expected behavior
<div [routerLink]="undefined">should not have tabIndex</div>

## Minimal reproduction of the problem with instructions

https://plnkr.co/edit/25H1bTPwqlJ33282MVh7?p=preview

## What is the motivation / use case for changing the behavior?
[Routerlink is supposed to now support null/undefined](6971). If it is null/undefined, it shouldn't impact the element at all. Having a tab index allows the user to focus on elements that shouldn't have focus

## Please tell us about your environment

<pre><code>
Angular version: 4.2.3

Browser:
- [X ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

### PR DESCRIPTION
**Pull request by harunurhan**
* Sets tabindex of non anchor routerLink host only if routerLink is not
null

Fixes #17616

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #17616 

## What is the new behavior?

It doesn't set `tabindex` for non anchor elements with `[routerLink]="null"`

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by harunurhan** [comment]
If wanted, I can remove the tabindex (only if it's `'0'`) when `routerLink` changes from not null value to `null`

**Review by mhevery — APPROVED** [review]

**Comment by atscott** [comment]
@mhevery - It seems to me that the tab index should be _removed_ if the link is set to `null` as well. This essentially only fixes the issue if the input is set to `null` on initialization but does not handle if the link was set to something else and then changes to `null`.

**Review by atscott — CHANGES_REQUESTED** [review]
In addition to https://github.com/angular/angular/pull/20922#issuecomment-755429485, the PR that addresses `tabindex` should also fix #28345 as well as consider what it means to have a `null` value for `routerLink` (#13980, #31154). Currently, a `null` value for `routerLink` is still a link to the current page, removing any fragment or query params.  That may not have been the original intent, but as it currently stands, the `tabindex` should still be set if the element is actually a link. If that's desired or unchangeable behavior, then disabling a link would be done with #31154 instead (and what does that mean for `tabindex`). All of these issues need to be solved in a coherent way.

Edit: for reference, the original PR to support `null`/`undefined` #13380 and the associated issue #6971

File: packages/router/src/directives/router_link.ts [hunk 1]
```diff
@@ -217,6 +214,16 @@ export class RouterLink implements OnChanges {
     }
   }
 
+  /** @nodoc */
+  ngOnChanges(changes: SimpleChanges) {
+    const routerLinkChange = changes['routerLink'];
+    if (routerLinkChange) {
+      if (this.tabIndex == null && routerLinkChange.currentValue != null) {
```
Comment (Totati) [inline packages/router/src/directives/router_link.ts:221]: Hi, I think, an input change from a value to null should be handled too.

**Comment by atscott** [comment]
Closing per https://github.com/angular/angular/pull/20922#pullrequestreview-562893089. This PR is incomplete and we also need a unified approach that considers what this means for accessibility standpoint.

**Pull request closed without merging**